Detections/SecurityEvent/RDP_Nesting.yaml

```diff
@@ -21,35 +21,38 @@ query: |
 
   let endtime = 1d;
   let starttime = 8d;
-  // The threshold below excludes matching on computer counts by account less than 5 in a given day.  Change the threshold as needed.
+  // The threshold below excludes matching on RDP connection computer counts of 5 or more by a given account and IP in a given day.  Change the threshold as needed.
   let threshold = 5;
   SecurityEvent
   | where TimeGenerated >= ago(endtime) 
   | where EventID == 4624 and LogonType == 10
   // Labeling the first RDP connection time, computer and ip
-  | extend FirstHop = TimeGenerated, FirstComputer = Computer, FirstIPAddress = IpAddress 
-  | join (
+  | extend FirstHop = TimeGenerated, FirstComputer = toupper(Computer), FirstIPAddress = IpAddress, Account = tolower(Account)  
+  | join kind=inner (
   SecurityEvent
   | where TimeGenerated >= ago(endtime) 
   | where EventID == 4624 and LogonType == 10
   // Labeling the second RDP connection time, computer and ip
-  | extend SecondHop = TimeGenerated, SecondComputer = Computer, SecondIPAddress = IpAddress 
+  | extend SecondHop = TimeGenerated, SecondComputer = toupper(Computer), SecondIPAddress = IpAddress, Account = tolower(Account)
   ) on Account
   // Make sure that the first connection is after the second connection --> SecondHop > FirstHop
   // Then identify only RDP to another computer from within the first RDP connection by only choosing matches where the Computer names do not match --> FirstComputer != SecondComputer
   // Then make sure the IPAddresses do not match by excluding connections from the same computers with first hop RDP connections to multiple computers --> FirstIPAddress != SecondIPAddress
-  | where SecondHop > FirstHop and FirstComputer != SecondComputer and FirstIPAddress != SecondIPAddress
+  | where FirstComputer != SecondComputer and FirstIPAddress != SecondIPAddress and SecondHop > FirstHop
+  // where the second hop occurs within 30 minutes of the first hop
+  | where SecondHop <= FirstHop+30m
   | distinct Account, FirstHop, FirstComputer, FirstIPAddress, SecondHop, SecondComputer, SecondIPAddress, AccountType, Activity, LogonTypeName, ProcessName
-  // use left anti to exclude anything from the previous 7 days where the Account and IP has connected to more than 5 computers.
+  // use left anti to exclude anything from the previous 7 days where the Account and IP has connected 5 or more computers.
   | join kind=leftanti (
   SecurityEvent
   | where TimeGenerated >= ago(starttime) and TimeGenerated < ago(endtime) 
   | where EventID == 4624 and LogonType == 10
-  | summarize makeset(Computer), ComputerCount = dcount(Computer) by Account, IpAddress
-  // Connection count to computer by same account and IP to determine rarity
-  | where ComputerCount > threshold
+  | summarize makeset(Computer), ComputerCount = dcount(Computer) by bin(TimeGenerated, 1d), Account = tolower(Account), IpAddress
```

Comment:
make_set has replaced makeset so maybe consider updating this. (Makeset still works)

Reply:
yeah, Ashwin and I discussed this.  It hasn't replaced it in the Log Analytics framework, while it works, it gets redlined and it won't complete intellisense.  Until LA adopts it, I don't think we should be using it.